pkg/registry/core/pod/storage/eviction.go

```diff
@@ -203,6 +203,15 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
+// canIgnorePDB returns true for pod conditions that allow the pod to be deleted
+// without checking PDBs.
+func canIgnorePDB(pod *api.Pod) bool {
+	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed || pod.Status.Phase == api.PodPending {
```

Comment:
@kubernetes/sig-node-pr-reviews is `PodPending` the right check here, or do we need to drill down into more detailed aspects (e.g. a `Ready` condition of `True` - "the Pod is able to serve requests and should be added to the load balancing pools of all matching Services;")

Reply:
@liggitt I have another patch that is a bit more defensive:

https://github.com/kubernetes/kubernetes/pull/81175

> otherwise we could race evicting with the kubelet marking the pod as ready

Without some kind of distributed locking mechanism, I think we're already exposed to a race here.  We discussed this a bit more in my other PR.

Synopsis of other PR:  Only removing pending pods if we have enough healthy nodes (PDB's consider disrupted nodes, rather than healthy nodes, so we have to perform that calculation ourselves).  This should provide enough buffer against multiple transactions happening at once.

Reply:
the pod object we get on line 119 has a resourceVersion we can use as a precondition in the delete call on line 134 that would ensure we only delete the pod if no other changes were made to it in the meantime. that wasn't important before this PR because succeeded and failed are terminal states.

Reply:
if a change *was* made to the pod in the meantime, the delete call would get a conflict error, and this code could refetch the pod and retry the logic. The RetryOnConflict helper method is used for this in many places.

Reply:
@liggitt if we get a conflict error, perhaps we should leave retry up to the client?  Send back the 529 (or whatever code it is), and then the PDB logic will run anew.

How do we ensure the delete call gets this conflict?  I thought that was already built in already?

Reply:
> How do we ensure the delete call gets this conflict? I thought that was already built in already?

set the resourceVersion precondition in deletionOptions if we are making a decision based on the information in a particular revision of the pod (as this PR is introducing)